test/parallel/test-http2-server-destroy-before-shutdown.js

```diff
@@ -0,0 +1,31 @@
+// Flags: --expose-http2
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const assert = require('assert');
+const http2 = require('http2');
+
+const server = http2.createServer();
+
+// Test that ERR_HTTP2_INVALID_SESSION is thrown when a stream is destroyed
+// before calling stream.session.shutdown
+server.on('stream', common.mustCall((stream) => {
+  stream.session.destroy();
+  assert.throws(
+    () => stream.session.shutdown(),
+    common.expectsError({
+      code: 'ERR_HTTP2_INVALID_SESSION',
+      message: /^The session has been destroyed$/
```

Comment:
This probably doesn't need to be a RegExp. 

Also, for future reference, `common.expectsError` can take `() => stream.session.shutdown()` as the first argument and the error object as the 2nd. (No need for `assert.throws`.)

Reply:
For example:

```js
common.expectsError(
  () => stream.session.shutdown(),
  {
    code: 'ERR_HTTP2_INVALID_SESSION',
    type: Error,
    message: 'The session has been destroyed'
  }
);
```

Reply:
Done!